proxy/context.go

```diff
@@ -200,6 +201,17 @@ func (c *context) SetOutgoingHost(h string)            { c.outgoingHost = h }
 func (c *context) Metrics() filters.Metrics            { return c.metrics }
 func (c *context) Tracer() opentracing.Tracer          { return c.tracer }
 func (c *context) ParentSpan() opentracing.Span        { return c.parentSpan }
+
+func (c *context) FastCgiFilename() string {
+	if c.fastCgiFilename == "" {
+		return "index.php"
+	}
+
+	return c.fastCgiFilename
+}
+
+func (c *context) SetFastCgiFilename(f string) { c.fastCgiFilename = f }
```

Comment:
I think it would be better not to change the FilterContext (see travis errors), but use a default filename as you have and use filters to configure it.
Just add the filename into the stateBag in the request filter and dispatch in the proxy.